crates/neon-build/src/lib.rs

```diff
@@ -31,6 +31,75 @@ cfg_if! {
                 println!("cargo:rustc-cdylib-link-arg=/DELAYLOAD:node.exe");
             }
         }
+    } else if #[cfg(windows)] {
+        // ^ automatically not neon-sys
+        use std::io::{Error, ErrorKind, Result};
+        use std::process::Command;
+        use std::path::Path;
+
+        fn node_version() -> Result<String> {
+            let output = Command::new("node").arg("-v").output()?;
+            if !output.status.success() {
+                let hopefully_stack_trace = String::from_utf8(output.stderr)
+                    .unwrap_or_else(|_| "<subprocess output garbage>".to_string());
+                panic!("Could not download node.lib: {}", hopefully_stack_trace);
+            }
+            let stdout = String::from_utf8(output.stdout).map_err(|error| {
+                Error::new(ErrorKind::InvalidData, error)
+            })?;
+            Ok(stdout.trim().to_string())
+        }
+
+        fn download_node_lib(version: &str, arch: &str) -> Result<Vec<u8>> {
+            let script = r#"
+                var url = process.argv[1]
+                require("https").get(url, function (res) {
+                    res.pipe(process.stdout);
+                });
+            "#;
+            let url = format!("https://nodejs.org/dist/{version}/win-{arch}/node.lib", version = version, arch = arch);
+
+            let output = Command::new("node")
+                .arg("-e").arg(script)
+                .arg(url)
+                .output()?;
+
+            Ok(output.stdout)
+        }
+
+        /// Set up the build environment by setting Cargo configuration variables.
+        pub fn setup() {
+            // If the user specified a node.lib path, we do not need to download
+            if let Some(node_lib_path) = std::env::var_os("NEON_NODE_LIB") {
+                let node_lib_path = Path::new(&node_lib_path);
+                // Clearing the file name returns the root+directory name
+                let dir = node_lib_path.with_file_name("");
+                let basename = node_lib_path.file_stem().expect("Could not parse lib name from NEON_NODE_LIB. Does the path include the full file name?");
+
+                println!("cargo:rustc-link-search=native={}", dir.display());
+                // Littul hack to output the OsStr file stem
+                println!("cargo:rustc-link-lib={}", Path::new(basename).display());
+                return;
+            }
+
+            let version = node_version().expect("Could not determine Node.js version");
```

Comment:
Can we use the `npm_config_target` environment variable if it's set? That's what node-gyp/electron uses.